cpp/src/io/parquet/page_enc.cu

```diff
@@ -1444,20 +1404,15 @@ __global__ void __launch_bounds__(128) gpuEncodePageHeaders(EncPage *pages,
 
   uint32_t t = threadIdx.x;
 
-  if (t < sizeof(EncPage) / sizeof(uint32_t)) {
-    reinterpret_cast<uint32_t *>(&page_g)[t] =
-      reinterpret_cast<uint32_t *>(&pages[start_page + blockIdx.x])[t];
-  }
+  if (t == 0) page_g = pages[start_page + blockIdx.x];
   __syncthreads();
-  if (t < sizeof(EncColumnChunk) / sizeof(uint32_t)) {
-    reinterpret_cast<uint32_t *>(&ck_g)[t] =
-      reinterpret_cast<const uint32_t *>(&chunks[page_g.chunk_id])[t];
-  }
+
+  if (t == 0) ck_g = chunks[page_g.chunk_id];
   __syncthreads();
-  if (t < sizeof(EncColumnDesc) / sizeof(uint32_t)) {
-    reinterpret_cast<uint32_t *>(&col_g)[t] = reinterpret_cast<const uint32_t *>(ck_g.col_desc)[t];
-  }
+
+  if (t == 0) col_g = *ck_g.col_desc;
```

Comment:
These three can be combined into a single if block to reduce number of __syncthreads.